mycore-base/src/main/java/org/mycore/common/MCRSession.java

```diff
@@ -69,7 +69,7 @@ sealed public class MCRSession implements Cloneable permits MCRScopedSession {
     private static final URI DEFAULT_URI = URI.create("");
 
     /** A map storing arbitrary session data * */
-    private Map<Object, Object> map = new Hashtable<>();
+    private Map<Object, Object> map = new HashMap<>();
```

Comment:
Pretty sure that this need proper handling for multiple threads. Use ConcurrentHashMao here

Reply:
You are right ... I changed possible candidates for multithreaded access with ConcurrentHashMap
(MCRSession and MCRAccessControlSystem)

I'm not sure about the "data"-Map in MCREvent. That should not be an issue?

The remaining maps are only used within methods and should be safe.